pkgs/os-specific/darwin/binutils/default.nix

```
@@ -49,7 +48,20 @@ stdenv.mkDerivation {
         ln -sv "$path" "$dest_path"
       done
     done
-  '';
+  ''
+  + (if stdenv.isAarch64 then ''
+    # On aarch64-darwin we must use clang, because "as" from cctools just doesn't
+    # handle the arch. Proxying calls via cctools to a wrapped clang produces
+    # quite a bit of warnings, so we're skipping the intermediary here.
+    makeWrapper "${clang-unwrapped}/bin/clang" "$out/bin/as" \
```

```suggestion
    rm "$out/bin/as"
    makeWrapper "${clang-unwrapped}/bin/clang" "$out/bin/as" \
```
And get rid of changes on x86_64-darwin?

---

Oh, this way there's no change for `x86_64-darwin` at all, I like it. Amended.

I also extracted the comment outside of the shell command to avoid rebuilds after comment chane.